### PR TITLE
Fiji - Fix the incorrect navigation

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes #BLUI-4723 .

Issue [275](https://github.com/etn-ccis/blui-react-native-workflows/issues/275)

#### Changes proposed in this Pull Request:

- Moved cleanup function to a new useEffect that has no dependencies in SelfRegistrationPager component.

#### Screenshots / Screen Recording (if applicable)

https://github.com/etn-ccis/blui-react-native-workflows/assets/133877691/a19d150f-6a5b-40f4-b57b-ec204eeeb799

#### To Test:

- yarn start:example-android/ios

#### Any specific feedback you are looking for?

-

### CODE REVIEW

**Review by ektaghag-eaton — APPROVED** [review]

**Review by surajeaton — COMMENTED** [review]
Please update the changelog.

File: login-workflow/example/android/app/build.gradle
Comment (surajeaton) [inline login-workflow/example/android/app/build.gradle:1]: Please revert the changes of this file.

File: login-workflow/example/android/app/build.gradle [hunk 1]
```diff
@@ -79,7 +79,7 @@ import org.apache.tools.ant.taskdefs.condition.Os
  */
 
 project.ext.react = [
-    enableHermes: true,  // clean and rebuild if changing
```
Comment (surajeaton) [inline login-workflow/example/android/app/build.gradle:82]: Please revert the changes of this file.

File: login-workflow/example/android/build.gradle [hunk 2]
```diff
@@ -4,8 +4,8 @@ buildscript {
     ext {
         buildToolsVersion = "31.0.0"
         minSdkVersion = 21
-        compileSdkVersion = 31
```
Comment (surajeaton) [inline login-workflow/example/android/build.gradle:7]: Please revert the changes of this file.

File: login-workflow/src/screens/SelfRegistrationPager.tsx [hunk 3]
```diff
@@ -203,14 +203,16 @@ export const SelfRegistrationPager: React.FC<SelfRegistrationPagerProps> = (prop
...
         }
-        return (): void => {
-            registrationActions.dispatch(RegistrationActions.requestRegistrationCodeReset());
-            registrationActions.dispatch(RegistrationActions.validateUserRegistrationReset());
-            registrationActions.dispatch(RegistrationActions.registerUserReset());
-        };
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [hasAcknowledgedError]);
 
+    useEffect(() => {
```
Comment (surajeaton) [inline login-workflow/src/screens/SelfRegistrationPager.tsx:209]: The suggested solution in the issue is to handle cleanup funtions `return (): void => { // cleanup functions}` in the separate useEffect hook with no dependencies.
This is not a cleanup as there is no `return`. Please add:
`return (): void => {
            registrationActions.dispatch(RegistrationActions.requestRegistrationCodeReset());
            registrationActions.dispatch(RegistrationActions.validateUserRegistrationReset());
            registrationActions.dispatch(RegistrationActions.registerUserReset());
        };`
        inside useEffect.

File: login-workflow/CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 -   Ability to hide the CreatePassword screen.
 -   Ability to hide the login form in the Login screen.
 
+### Fixed
+
+-   Incorrect navigation when error occured ([#275](https://github.com/etn-ccis/blui-react-native-workflows/issues/275)).
```
Comment (surajeaton) [inline login-workflow/CHANGELOG.md:17]: Need to mention on which screen is it happening:

Incorrect navigation on dismissal of error occurred in Account Details screen

**Review by surajeaton — APPROVED** [review]
Looks good.
